### PR TITLE
Added transip_global_key to config.yaml

### PR DESCRIPTION
**Pull request by Goopher**
## Summary by CodeRabbit

- **Documentation**
	- Updated installation and usage instructions for the Letsencrypt add-on.
	- Expanded details on obtaining certificates and supported DNS providers.
	- Added examples for various DNS challenges, including Cloudflare, GoDaddy, and Google Cloud DNS.
	- Emphasized security best practices, including the use of API tokens.

- **New Features**
	- Introduced a new configuration field, `transip_global_key`, for the TransIP DNS provider in the configuration options.
	- Updated to version 5.2.10 in the configuration file.

### CODE REVIEW

**Comment by agners** [comment]
Hm, I see this got missed in #3835. Generally LGTM, but this needs a config.yaml version bump and a CHANGELOG.md entry.

File: letsencrypt/config.yaml [hunk 1]
```diff
@@ -120,6 +120,7 @@ schema:
     sakuracloud_api_token: str?
     transip_api_key: str?
     transip_username: str?
+    transip_global_key: str?
```
Comment (nathansamson) [inline letsencrypt/config.yaml:123]: Probably better to define as `list(yes|no)?`

Comment (Goopher) [reply letsencrypt/config.yaml]: Good point, fixed

**Comment by Goopher** [comment]
> Hm, I see this got missed in #3835. Generally LGTM, but this needs a config.yaml version bump and a CHANGELOG.md entry.

@agners Done

**Review by agners — APPROVED** [review]
Thanks, LGTM!

**Comment by rdoubleu** [comment]
Thanks guys, But the issue is still there? of do i something wrong?

after the update i rund it again. but still the same issue. version 5.2.10

this is what i fill in in the Yaml.

```
dns:
  provider: dns-transip
  transip_username: username
  dns_transip_global_key: "yes"
  transip_api_key: |
    -----BEGIN PRIVATE KEY-----
.....
    -----END PRIVATE KEY-----
```

this is my log. 

```
cont-init: info: /etc/cont-init.d/file-structure.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun lets-encrypt (no readiness notification)
s6-rc: info: service legacy-services successfully started
[13:23:17] INFO: Selected DNS Provider: dns-transip
[13:23:17] INFO: Use propagation seconds: 60
[13:23:17] INFO: Increasing DNS propagation limit for TransIP to at least 240 seconds.
[13:23:17] INFO: Detecting existing certificate type for home.rdoubleu.nl
Saving debug log to /var/log/letsencrypt/letsencrypt.log
[13:23:21] INFO: No certificate found - using 'ecdsa' key type.
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Requesting a certificate for my.domain.nl
Encountered exception during recovery: ValueError: dns_transip_global_key should have either 'yes' or 'no' as value
An unexpected error occurred:
ValueError: dns_transip_global_key should have either 'yes' or 'no' as value
Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /var/log/letsencrypt/letsencrypt.log or re-run Certbot with -v for more details.
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

**Comment by Goopher** [comment]
@rdoubleu can you try transip_global_key instead of dns_transip_global_key?

**Comment by rdoubleu** [comment]
> @rdoubleu can you try transip_global_key instead of dns_transip_global_key?

Done. this was the trick!

**Comment by Goopher** [comment]
@nathansamson Yup, I think cause we're setting the variable always, even when nothing is present in the bashio::config.

I think replacing the variable assignment by this would fix it:

`if bashio::config.has_value 'dns.transip_global_key'; then
    dns_transip_global_key=$(bashio::config 'dns.transip_global_key')
fi`

Let me setup a PR

**Comment by nathansamson** [comment]
> @nathansamson Yup, I think cause we're setting the variable always, even when nothing is present in the bashio::config.
> 
> I think replacing the variable assignment by this would fix it:
> 
> `if bashio::config.has_value 'dns.transip_global_key'; then dns_transip_global_key=$(bashio::config 'dns.transip_global_key') fi`
> 
> Agreed?

In all honesty, this was my first time diving into a homeassistant plugin. But from what I can see from the other code, i would say yes that would do it...

**Comment by hapklaar** [comment]
Where can I find documentation about this key, what it does and when to use what value? Not setting it breaks the integration, should be a default value or notification it is a breaking change.

**Pull request merged**